tss-esapi/src/structures/buffers/private.rs

```diff
@@ -0,0 +1,70 @@
+use crate::{
+    traits::{Marshall, UnMarshall},
+    ReturnCode,
+};
+use std::convert::TryInto;
+use tss_esapi_sys::_PRIVATE;
+
+buffer_type!(Private, ::std::mem::size_of::<_PRIVATE>(), TPM2B_PRIVATE);
+
+impl Marshall for Private {
+    const BUFFER_SIZE: usize = std::mem::size_of::<TPM2B_PRIVATE>();
+
+    /// Produce a marshalled [`TPM2B_PRIVATE`]
+    fn marshall(&self) -> Result<Vec<u8>> {
```

Comment:
I agree with @Superhepper - the `_offset` version of the trait methods are better suited here, as `marshall` and `unmarshall` are already implemented by default if the `_offset` version exists.